src/connector.rs

```diff
@@ -0,0 +1,593 @@
+        sdk::assert_one_yocto();
+        let args =
+            TransferCallCallArgs::try_from_slice(&sdk::read_input()).expect(ERR_FAILED_PARSE);
+        #[cfg(feature = "log")]
+        sdk::log(&format!(
+            "Transfer call to {} amount {}",
+            args.receiver_id, args.amount,
+        ));
+
+        self.ft
+            .ft_transfer_call(&args.receiver_id, args.amount, &args.memo, args.msg);
+    }
+
+    /// FT storage deposit logic
+    pub fn storage_deposit(&mut self) {
+        let args =
+            StorageDepositCallArgs::try_from_slice(&sdk::read_input()).expect(ERR_FAILED_PARSE);
+        let res = self
+            .ft
+            .storage_deposit(args.account_id.as_ref(), args.registration_only)
+            .try_to_vec()
+            .unwrap();
+        self.save_contract();
+        sdk::return_output(&res[..]);
+    }
+
+    /// FT storage withdraw
+    pub fn storage_withdraw(&mut self) {
+        sdk::assert_one_yocto();
+        let args =
+            StorageWithdrawCallArgs::try_from_slice(&sdk::read_input()).expect(ERR_FAILED_PARSE);
+        let res = self.ft.storage_withdraw(args.amount).try_to_vec().unwrap();
+        self.save_contract();
+        sdk::return_output(&res[..]);
+    }
+
+    /// Get balance of storage
+    pub fn storage_balance_of(&self) {
+        let args =
+            StorageBalanceOfCallArgs::try_from_slice(&sdk::read_input()).expect(ERR_FAILED_PARSE);
+        let res = self
+            .ft
+            .storage_balance_of(&args.account_id)
+            .try_to_vec()
+            .unwrap();
+        sdk::return_output(&res[..]);
+    }
+
+    /// ft_on_transfer callback function
+    pub fn ft_on_transfer(&mut self, engine: &Engine) {
+        #[cfg(feature = "log")]
+        sdk::log("Call ft_on_trasfer");
+        let args = FtOnTransfer::try_from_slice(&sdk::read_input()).expect(ERR_FAILED_PARSE);
+        let predecessor_account_id = String::from_utf8(sdk::predecessor_account_id()).unwrap();
+        let current_account_id = String::from_utf8(sdk::current_account_id()).unwrap();
+        // Parse message with specific rules
+        let message_data = self.parse_on_transfer_message(&args.msg);
+
+        // Special case when predecessor_account_id is current_account_id
+        if current_account_id == predecessor_account_id {
+            let fee = message_data.fee.as_u128();
+            self.burn_near(current_account_id, args.amount);
+            self.mint_eth(message_data.recipient, args.amount - fee);
+            // Mint fee to relayer
+            if fee > 0 {
+                let evm_relayer_address: EthAddress = engine
+                    .get_relayer(&message_data.relayer.as_bytes())
+                    .expect("ERR_WRONG_RELAYER_ID")
+                    .0;
+                self.mint_eth(evm_relayer_address, fee);
+            }
+        } else {
+            // Implement new scheme when PR #62 is merged
+            todo!();
```

Comment:
Just commenting to call this out. #62 is now merged, so this should be updated according to the comment.

Reply:
I don't think this is resolved, as its still a todo?